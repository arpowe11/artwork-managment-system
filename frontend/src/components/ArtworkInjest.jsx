import React, { useState } from "react";
import toast from "react-hot-toast";
import Navigation from "./Navigation";
import ingestStyles from "../assets/styles/ingest-styles";
import { useNavigate } from "react-router-dom";

const ArtworkInjest = () => {
    const [loading, setLoading] = useState(false);
    const navigate = useNavigate();

    const handleInjest = async () => {
        try {
            setLoading(true);
            toast.loading("Ingest started...");

            const response = await fetch("http://localhost:5001/api/ingest", { // TODO: Python api deploy uri
                method: "POST",
                headers: { "Content-Type": "application/json" },
                body: JSON.stringify({}),
            });

            if (!response.ok) throw new Error(`Server error: ${response.status}`);

            const result = await response.json();
            toast.dismiss(); // remove loading toast
            toast.success(result.message || "Ingestion completed!");
        } catch (err) {
            toast.dismiss();
            toast.error("An error occured during injestion.");
        } finally {
            setLoading(false);
        }
    };

    const handleClick = (e) => {
        e.preventDefault();
        navigate("/artwork-manager");
    }

    return (
        <div style={ingestStyles.container}>
            <Navigation />
            <h1 style={ingestStyles.header}>Artwork Ingest Page</h1>
            <p style={ingestStyles.paragraph}>
                To get anything new from the Artwork API, click the button below.
                To add your own artwork navigate to <a href="#" onClick={handleClick}> here </a>
            </p>
            <button
                onClick={handleInjest}
                style={ingestStyles.button}
                onMouseEnter={(e) => {
                    e.target.style.background = ingestStyles.buttonHover.background;
                    e.target.style.transform = ingestStyles.buttonHover.transform;
                }}
                onMouseLeave={(e) => {
                e.target.style.background = ingestStyles.button.background;
                e.target.style.transform = "scale(1)";
                }}
            >
                Ingest Data
            </button>
        </div>

    );
}

export default ArtworkInjest;