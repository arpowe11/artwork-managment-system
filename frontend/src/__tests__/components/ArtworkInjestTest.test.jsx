import { describe, it, vi, expect, afterEach } from "vitest";
import { render, screen, cleanup } from "@testing-library/react";
import '@testing-library/jest-dom/vitest';
import React from "react";
import { MemoryRouter } from "react-router-dom";
import ArtworkInjest from "../../components/ArtworkInjest";


// Mock useNavigate
const mockedNavigate = vi.fn();

vi.mock("react-router-dom", async () => {
  const actual = await vi.importActual("react-router-dom");
  return {
    ...actual,
    useNavigate: () => mockedNavigate,
  };
});

afterEach(() => {
  cleanup();
  mockedNavigate.mockReset();
});

describe("ArtworkInjest", () => {
    it("should render the heading", () => {
        render(
            <MemoryRouter>
                <ArtworkInjest />
            </MemoryRouter>
        );

        const heading = screen.getByRole("heading", { name: /artwork ingest page/i });
        expect(heading).toBeInTheDocument();
    });

    it("should render the ingest button", () => {
        render(
            <MemoryRouter>
                <ArtworkInjest />
            </MemoryRouter>
        );

        const button = screen.getByRole("button", { name: /ingest data/i });
        expect(button).toBeInTheDocument();
        expect(button).toHaveTextContent(/ingest data/i);
    });

    it("should render the link", () => {
        render(
            <MemoryRouter>
                <ArtworkInjest />
            </MemoryRouter>
        );

        const link = screen.getByRole("link", { name: /here/i });
        expect(link).toBeInTheDocument();
        expect(link).toHaveTextContent(/here/i);
    });
});
