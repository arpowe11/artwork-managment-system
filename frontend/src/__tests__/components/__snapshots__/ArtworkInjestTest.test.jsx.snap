// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ArtworkInjest Component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="min-height: 100vh; margin: 0px; padding: 2rem; text-align: center; background: linear-gradient(to right, #4facfe, #00f2fe); display: flex; flex-direction: column; justify-content: flex-start; align-items: center;"
  >
    <div>
      <nav
        style="position: fixed; top: 0px; left: 0px; width: 100%; display: flex; justify-content: space-between; align-items: center; padding: 0px 2rem; height: 60px; background-color: rgb(51, 51, 51); z-index: 1000; box-sizing: border-box;"
      >
        <div
          style="display: flex; gap: 2rem;"
        >
          <a
            href="http://localhost:5173/dashboard"
            style="color: rgb(255, 255, 255); text-decoration: none; font-size: 1.1rem; transition: color 0.3s ease;"
          >
            Home
          </a>
          <a
            href="http://localhost:5173/artwork-injest"
            style="color: rgb(255, 255, 255); text-decoration: none; font-size: 1.1rem; transition: color 0.3s ease;"
          >
            Artwork Ingest
          </a>
          <a
            href="http://localhost:5173/artwork-manager"
            style="color: rgb(255, 255, 255); text-decoration: none; font-size: 1.1rem; transition: color 0.3s ease;"
          >
            Artwork Manager
          </a>
        </div>
        <div
          style="display: flex; align-items: center;"
        >
          <button
            style="padding: 10px 24px; margin-top: 20px; cursor: pointer; background-color: rgb(230, 57, 70); color: rgb(255, 255, 255); border-radius: 8px; font-size: 1rem; font-weight: 600; transition: all 0.3s ease; box-shadow: 0 4px 8px rgba(0, 0, 0, 0.15);"
          >
            Logout
          </button>
        </div>
      </nav>
    </div>
    <h1
      style="font-size: 2rem; margin-top: 3rem; margin-bottom: 2rem; color: rgb(255, 255, 255); text-shadow: 1px 1px 4px rgba(0,0,0,0.2);"
    >
      Artwork Ingest Page
    </h1>
    <p
      style="font-size: 1rem; color: rgb(68, 68, 68); margin-bottom: 2rem; max-width: 600px; line-height: 1.5;"
    >
      To get anything new from the Artwork API, click the button below. To add your own artwork navigate to 
      <a
        href="#"
      >
         here 
      </a>
    </p>
    <button
      style="padding: 0.8rem 1.5rem; font-size: 1rem; font-weight: bold; background: rgb(80, 82, 80); color: rgb(255, 255, 255); border-radius: 8px; cursor: pointer; transition: all 0.2s ease;"
    >
      Ingest Data
    </button>
  </div>
</DocumentFragment>
`;
